packages/components/src/button/index.js

```diff
@@ -19,18 +19,76 @@ import { VisuallyHidden } from '../visually-hidden';
 
 const disabledEventsOnDisabledButton = [ 'onMouseDown', 'onClick' ];
 
+// TODO: add JSDOCS
+function useDeprecatedProps( {
+	isDefault,
+	isPrimary,
+	isSecondary,
+	isTertiary,
+	isLink,
+	variant,
+	...otherProps
+} ) {
+	let computedVariant = variant;
+
+	if ( isPrimary ) {
+		deprecated( 'Button isPrimary prop', {
+			since: '5.9',
+			alternative: 'variant="primary"',
+		} );
+
+		computedVariant ??= 'primary';
+	}
+
+	if ( isTertiary ) {
+		deprecated( 'Button isTertiary prop', {
+			since: '5.9',
+			alternative: 'variant="tertiary"',
+		} );
+
+		computedVariant ??= 'tertiary';
+	}
+
+	if ( isSecondary ) {
+		deprecated( 'Button isSecondary prop', {
+			since: '5.9',
+			alternative: 'variant="secondary"',
+		} );
+
+		computedVariant ??= 'secondary';
+	}
+
+	if ( isDefault ) {
+		deprecated( 'Button isDefault prop', {
+			since: '5.4',
+			alternative: 'variant="secondary"',
+		} );
+
+		computedVariant ??= 'secondary';
+	}
+
+	if ( isLink ) {
+		deprecated( 'Button isLink prop', {
+			since: '5.9',
+			alternative: 'variant="link"',
+		} );
+
+		computedVariant ??= 'link';
+	}
```

Comment:
Not sure about the `isDefault` => `secondary` and `isLink` => `link` mappings.

This previous PR from @sarayourfriend was instead mapping both deprecated props to `plain`: https://github.com/WordPress/gutenberg/pull/29990/files#diff-e931867515d855b923dc4148f4e2f9277bf7b37cd766f95e28ac11df4eb0b375R10-R54

Reply:
We decided to keep the code as-is currently — `isDefault` mapped to `secondary` and `undefined` as default / fallback value